examples/search/xgb.py

```diff
@@ -0,0 +1,87 @@
+import pickle
```

Comment:
We need a docstring up here similar to the rest of the examples. Something like:
```
"""
======================================
Train a classifier and regressor with xgb and sk-dist
======================================

Something, something, motivation for this example

Example output run:

-- Grid Search --
Best Score: 0.9936882800963308
Best colsample_bytree: 1.0
Best learning_rate: 0.05
Best max_depth: 4
Best n_estimators: 300
DistGridSearchCV(cv=5, error_score='raise-deprecating',
                 estimator=XGBClassifier(base_score=0.5, booster='gbtree',
                                         colsample_bylevel=1,
                                         colsample_bynode=1, colsample_bytree=1,
                                         gamma=0, learning_rate=0.1,
                                         max_delta_step=0, max_depth=3,
                                         min_child_weight=1, missing=nan,
                                         n_estimators=100, n_jobs=1,
                                         nthread=None,
                                         objective='binary:logistic',
                                         random_state=0, reg_alpha=0,
                                         reg_lambda=1, scale_pos_weight=1,
                                         seed=None, silent=None, subsample=1,
                                         verbosity=1),
                 iid='warn', n_jobs=None,
                 param_grid={'colsample_bytree': [0.6, 0.8, 1.0],
                             'learning_rate': [0.05, 0.01],
                             'max_depth': [4, 6, 8],
                             'n_estimators': [100, 200, 300]},
                 partitions='auto', pre_dispatch='2*n_jobs', preds=False,
                 refit=True, return_train_score=False, sc=None,
                 scoring='roc_auc', verbose=0)
-- Grid Search --
Best Score: -18.452273211144295
Best colsample_bytree: 0.8
Best learning_rate: 0.05
Best max_depth: 4
Best n_estimators: 200
DistGridSearchCV(cv=5, error_score='raise-deprecating',
                 estimator=XGBRegressor(base_score=0.5, booster='gbtree',
                                        colsample_bylevel=1, colsample_bynode=1,
                                        colsample_bytree=1, gamma=0,
                                        importance_type='gain',
                                        learning_rate=0.1, max_delta_step=0,
                                        max_depth=3, min_child_weight=1,
                                        missing=nan, n_estimators=100, n_jobs=1,
                                        nthread=None,
                                        objective='reg:squarederror',
                                        random...
                                        reg_lambda=1, scale_pos_weight=1,
                                        seed=None, silent=None, subsample=1,
                                        verbosity=1),
                 iid='warn', n_jobs=None,
                 param_grid={'colsample_bytree': [0.6, 0.8, 1.0],
                             'learning_rate': [0.05, 0.01],
                             'max_depth': [4, 6, 8],
                             'n_estimators': [100, 200, 300]},
                 partitions='auto', pre_dispatch='2*n_jobs', preds=False,
                 refit=True, return_train_score=False, sc=None,
                 scoring='neg_mean_squared_error', verbose=0)
"""
print(__doc__)
```